charmcraft/commands/version.py

```diff
@@ -29,35 +30,4 @@ class VersionCommand(BaseCommand):
 
     def run(self, parsed_args):
         """Run the command."""
-        # XXX: we need to define how we want to store the project version (in config/file/etc.)
-        version = '0.0.1'
-        logger.info('Version: %s', version)
-
-
-# XXX: the following commands are left here as an example of different commands... but in
-# reality new commands will have new files for them
-
-
-class CommandExampleDebug(BaseCommand):
-    """Just an example."""
-
-    name = 'example-debug'
-    help_msg = "show msg in debug"
-
-    def fill_parser(self, parser):
-        parser.add_argument('foo')
-        parser.add_argument('--bar', action='store_true', help="To use this command in a bar")
-
-    def run(self, parsed_args):
-        logger.debug(
-            "Example showing log in DEBUG: foo=%s bar=%s", parsed_args.foo, parsed_args.bar)
-
-
-class CommandExampleError(BaseCommand):
-    """Just an example."""
-
-    name = 'example-error'
-    help_msg = "show msg in error"
-
-    def run(self, parsed_args):
-        logger.error("Example showing log in ERROR.")
+        logger.info('Version: %s', __version__)
```

Comment:
I really appreciate this, many tools have come to have this added, as it really aids in debugging when someone is reporting an issue.
In Juju we went as far as updating the build script to embed "git describe --dirty", which gives a version string and the exact hash, and a field to tell you if it might be including local changes as well.

Reply:
Great! We could discuss in the future more specific versions, indeed!